services-api/src/main/java/io/scalecube/services/ServiceFactory.java

```diff
@@ -0,0 +1,36 @@
+package io.scalecube.services;
+
+import java.util.Collection;
+import reactor.core.publisher.Mono;
+
+/** Manages the life cycle of all services registered with Scale Cube Services. */
+public interface ServiceFactory {
+
+  /**
+   * Provide service definitions.
+   *
+   * @param microservices scale cube instance.
+   * @return collection of service definitions - service type and tags.
+   * @see ServiceDefinition
+   */
+  Mono<? extends Collection<ServiceDefinition>> getServiceDefinitions(
```

Comment:
Why it's a Mono here? What a usecase?

Reply:
Lazy init spring context?

Reply:
@artem-v are right, we don't need to have Mono in this interface at all. If you need such delusion as lazy initialization you can provide it at level up.

Reply:
What if we're reading from the disk?

Reply:
What are the arguments against?

Reply:
> What if we're reading from the disk?

We don't take in account theoretical arguments. Again. We don't need to async mode to read a dozens of tuples Class+HasMap